modules/apigee_edge_actions/tests/src/Kernel/Plugin/RulesEvent/EdgeEntityEventTestBase.php

```diff
@@ -0,0 +1,172 @@
+    'apigee_edge_actions_debug',
+    'apigee_edge',
+    'apigee_mock_api_client',
+    'dblog',
+    'key',
+    'options',
+  ];
+
+  /**
+   * The entity storage for Rules config entities.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $storage;
+
+  /**
+   * The user account.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $account;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->storage = $this->container->get('entity_type.manager')->getStorage('rules_reaction_rule');
+
+    $this->installConfig(['apigee_edge']);
+    $this->installEntitySchema('user');
+    $this->installSchema('dblog', ['watchdog']);
+    $this->installSchema('system', ['sequences']);
+    $this->installSchema('user', ['users_data']);
+
+    $this->baseSetUp();
+
+    /** @var \Drupal\user\UserInterface $account */
+    $this->account = User::create([
+      'mail' => $this->randomMachineName() . '@example.com',
+      'name' => $this->randomMachineName(),
+      'first_name' => $this->getRandomGenerator()->word(16),
+      'last_name' => $this->getRandomGenerator()->word(16),
+    ]);
+    $this->account->save();
+    $this->queueDeveloperResponse($this->account, Response::HTTP_CREATED);
+  }
+
+  /**
+   * Helper to create a DeveloperApp entity.
+   *
+   * @return \Drupal\apigee_edge\Entity\DeveloperAppInterface
+   *   A DeveloperApp entity.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function createDeveloperApp(): DeveloperAppInterface {
+    /** @var \Drupal\apigee_edge\Entity\DeveloperAppInterface $entity */
+    $entity = DeveloperApp::create([
+      'appId' => 1,
+      'name' => $this->randomMachineName(),
+      'status' => App::STATUS_APPROVED,
+      'displayName' => $this->randomMachineName(),
+    ]);
+    $entity->setOwner($this->account);
+    $this->queueDeveloperAppResponse($entity);
+    $entity->save();
+
+    return $entity;
+  }
+
+  /**
+   * Helper to create a Team entity.
+   *
+   * @return \Drupal\apigee_edge_teams\Entity\TeamInterface
+   *   A Team entity.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function createTeam(): TeamInterface {
+    /** @var \Drupal\apigee_edge_teams\Entity\TeamInterface $team */
+    $team = Team::create([
+      'name' => $this->randomMachineName(),
+      'displayName' => $this->randomGenerator->name(),
+    ]);
+    $this->queueCompanyResponse($team->decorated());
+    $this->queueDeveloperResponse($this->account);
+    $team->save();
+
+    return $team;
+  }
+
+  /**
+   * Helper to add Edge entity response to stack.
+   *
+   * @param \Drupal\apigee_edge\Entity\EdgeEntityInterface $entity
+   *   The Edge entity.
+   */
+  protected function queueDeveloperAppResponse(EdgeEntityInterface $entity) {
+    $this->stack->queueMockResponse([
+      'get_developer_app' => [
+        'app' => $entity,
+      ],
+    ]);
+  }
```

Comment:
Can we move these helper methods into `\Drupal\Tests\apigee_mock_api_client\Traits\ApigeeMockApiClientHelperTrait`?

Reply:
Done.